src/Client/ClientBase.cpp

```diff
@@ -1030,6 +1035,9 @@ void ClientBase::initKeystrokeInterceptor()
     {
         keystroke_interceptor = std::make_unique<TerminalKeystrokeInterceptor>(stdin_fd, error_stream);
         keystroke_interceptor->registerCallback(' ', [this]() { progress_table_toggle_on = !progress_table_toggle_on; });
+
+        if (isEmbeeddedClient())
+            keystroke_interceptor->registerCallback(0x03, [this]() { tryStopQuery(); });
```

Comment:
Should it be unregistered?

Reply:
I think no, the `keystroke_interceptor` is a field of a Client, not a global object, so it will be destroyed after session is closed.

Reply:
Found `stopKeystrokeInterceptorIfExists` that is called after query executed, so indeed, should be OK.